tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/CheckAllTool/ChangelogValidationTool.cs

```diff
@@ -0,0 +1,208 @@
+        private readonly IOutputService output;
+        private readonly IGitHelper gitHelper;
+
+        private readonly Option<string> packagePathOption = new(["--package-path", "-p"], "Path to the package directory to check") { IsRequired = true };
+
+        public ChangelogValidationTool(ILogger<ChangelogValidationTool> logger, IOutputService output, IGitHelper gitHelper) : base()
+        {
+            this.logger = logger;
+            this.output = output;
+            this.gitHelper = gitHelper;
+            CommandHierarchy = [SharedCommandGroups.Checks];
+        }
+
+        public override Command GetCommand()
+        {
+            Command command = new("changelog-validation", "Run changelog validation for SDK projects");
+            command.AddOption(packagePathOption);
+            command.SetHandler(async ctx => { await HandleCommand(ctx, ctx.GetCancellationToken()); });
+            return command;
+        }
+
+        public override async Task HandleCommand(InvocationContext ctx, CancellationToken ct)
+        {
+            var packagePath = ctx.ParseResult.GetValueForOption(packagePathOption);
+            var result = await RunChangelogValidation(packagePath);
+
+            output.Output(result);
+            ctx.ExitCode = ExitCode;
+        }
+
+        [McpServerTool(Name = "RunChangelogValidation"), Description("Run changelog validation for SDK packages. Provide absolute path to package root as param.")]
+        public async Task<ICLICheckResponse> RunChangelogValidation(string packagePath)
+        {
+            var stopwatch = System.Diagnostics.Stopwatch.StartNew();
+            
+            try
+            {
+                logger.LogInformation($"Starting changelog validation for package at: {packagePath}");
+                
+                if (!Directory.Exists(packagePath))
+                {
+                    SetFailure(1);
+                    return new FailureCLICheckResponse(1, "", $"Package path does not exist: {packagePath}");
+                }
+
+                // Find the SDK repository root by looking for common repository indicators
+                // Start from the package path and work upwards to find the SDK repo root
+                var packageRepoRoot = gitHelper.DiscoverRepoRoot(packagePath);
+                if (string.IsNullOrEmpty(packageRepoRoot))
+                {
+                    SetFailure(1);
+                    return new FailureCLICheckResponse(1, "", $"Could not find repository root from package path: {packagePath}");
+                }
+
+                // Construct the path to the PowerShell script in the SDK repository
+                // The script should be in the package's repository root, not relative to this tool's location
+                var scriptPath = Path.Combine(packageRepoRoot, "eng", "common", "scripts", "Verify-ChangeLog.ps1");
+                
+                if (!File.Exists(scriptPath))
```

Comment:
The file helper has file exists validation now if you'd like to use a more robust check implementation.

Reply:
I just see validateEmptyDirectory, is that what you are referring to?